zipline/pipeline/loaders/frame.py

```diff
@@ -174,7 +175,13 @@ def load_adjusted_array(self, columns, dates, assets, mask):
             column: AdjustedArray(
                 # Pull out requested columns/rows from our baseline data.
                 data=data,
-                adjustments=self.format_adjustments(dates, assets),
+                adjustments=self.format_adjustments(dates, sids),
                 missing_value=column.missing_value,
             ),
         }
+
+    def _validate_input_column(self, column):
+        """Make sure a passed column is our column.
+        """
+        if column != self.column and column.unspecialize() != self.column:
```

Comment:
By unspecializing here, we allow a user to register a single DataFrame loader that can serve queries across multiple domains if the loader is constructed with a generic column. 

The `load_adjusted_array` implementation here does the right thing if you give it data for assets on multiple domains (it filters down to just the requested assets and calendars at call time). 

I think something like this will be necessary eventually to support dynamic domains, so given that this "just works" with what I think are the right semantics, I'm inclined to not force people to statically register a bunch of specializations if we don't need to.